### ISSUE TITLE
S3 list_objects doesn't respect --max-key and --starting-token

### ISSUE THREAD

**Issue by ten0s** [localstack/localstack#214]
Localstack implementation doesn't respect filtering and always returns all stored objects.

### Amazon S3 setup
```
$ export BUCKET=xyz_abcdef
$ export ENDPOINT=
$ aws $ENDPOINT s3 mb s3://$BUCKET
$ for i in `seq 1 10`; do echo hello$i | aws $ENDPOINT s3 cp - s3://$BUCKET/hello$i; done
$ aws $ENDPOINT s3api list-objects --bucket $BUCKET | jq '.Contents | length'
10
```
### OK
```
$ aws $ENDPOINT s3api list-objects --bucket $BUCKET --max-key 1 | jq '.Contents | length'
1
$ aws $ENDPOINT s3api list-objects --bucket $BUCKET --starting-token hello8 | jq '.Contents | length'
1
```

### Localstack setup
```
$ export BUCKET=xyz_abcdef
$ export ENDPOINT='--endpoint-url http://localhost:4572'
$ aws $ENDPOINT s3 mb s3://$BUCKET
$ for i in `seq 1 10`; do echo hello$i | aws $ENDPOINT s3 cp - s3://$BUCKET/hello$i; done
$ aws $ENDPOINT s3api list-objects --bucket $BUCKET | jq '.Contents | length'
10
```
### FAIL
```
$ aws $ENDPOINT s3api list-objects --bucket $BUCKET --max-key 1 | jq '.Contents | length'
10
$ aws $ENDPOINT s3api list-objects --bucket $BUCKET --starting-token hello8 | jq '.Contents | length'
10
```

Thank you!

**Comment by julienduchesne** [comment]
Hi @whummer. I'd like to start contributing to Localstack. Mind if I take on this issue?

**Comment by whummer** [comment]
Thanks @julienduchesne , looking forward to your contribution! Please let me know if you need any help with the implementation.

**Comment by julienduchesne** [comment]
I can't assign myself but it's no worries.

**Comment by julienduchesne** [comment]
I posted two pull requests on the moto repository. When they're merged, I'll post again here.

**Comment by julienduchesne** [comment]
@whummer. This is fixed in moto version 1.1.4:
https://github.com/spulec/moto/pull/1124
https://github.com/spulec/moto/pull/1123
